pkg/cli/mirror/publish_test.go

```diff
@@ -74,6 +74,7 @@ func Test_MetadataError(t *testing.T) {
 		if err != nil {
 			t.Error(err)
 		}
+		defer server.Close()
```

Comment:
Do `t.Cleanup(server.Close)` instead, and call it after `NewServer` in case `url.Parse` fails (which should have a `continue` in it's error branch).